infrastructure/networking.tf

```diff
@@ -166,3 +166,83 @@ resource "aws_lb_target_group_attachment" "api-https" {
   target_id = "${aws_instance.api_server_1.id}"
   port = 443
 }
+
+resource "aws_acm_certificate" "ssl-cert" {
+  domain_name = "staging.refine.bio"
+  validation_method = "DNS"
+
+  tags {
+    Environment = "data-refinery-circleci-staging"
+  }
+}
+
+
+resource "aws_cloudfront_distribution" "static-distribution" {
+  aliases = ["${var.static_bucket_prefix == "dev" ? var.user : var.static_bucket_prefix}${var.static_bucket_root}"]
+
+  origin {
+    domain_name = "${aws_s3_bucket.data-refinery-static.website_endpoint}"
+    origin_id = "data-refinery-circleci-staging"
+
+    custom_origin_config {
+      origin_protocol_policy = "http-only"
+      http_port = "80"
+      https_port = "443"
+      origin_ssl_protocols =  ["TLSv1.2", "TLSv1.1", "TLSv1"]
+    }
+  }
+
+  enabled = true
+  default_root_object = "index.html"
+
+  logging_config {
+    include_cookies = false
+    bucket = "${aws_s3_bucket.data-refinery-static-access-logs.bucket_domain_name}"
+  }
+
+  # This makes refreshing pages in nested paths work:
+  # https://stackoverflow.com/a/35673266/6095378
+  custom_error_response {
+    error_caching_min_ttl = 0
+    error_code = 403
+    response_code = 200
+    response_page_path = "/index.html"
+  }
+
+  default_cache_behavior {
+    allowed_methods  = ["DELETE", "GET", "HEAD", "OPTIONS", "PATCH", "POST", "PUT"]
+    cached_methods   = ["GET", "HEAD"]
+    target_origin_id = "data-refinery-${var.user}-${var.stage}"
+
+    forwarded_values {
+      query_string = false
+
+      cookies {
+        forward = "none"
+      }
+    }
+
+    viewer_protocol_policy = "allow-all" # Change this to redirect-to-https
+    # XXX: This might make things pricey, but what should they actually be?
```

Comment:
Let's get rid of this XXX and explain what this behavior does and what the possibilities here can be.